From 985cd10b8da1b3ee59ced7784b4252441baeceef Mon Sep 17 00:00:00 2001
From: Fabio Estevam <festevam@gmail.com>
Date: Fri, 20 Dec 2019 19:06:33 -0300
Subject: [PATCH] imx6qdl-ccimx6sb: Disable EDID support

Disable EDID from the native HDMI port as well as from
the ADV7513 to workaround a kernel hang when the Albacore
board is booted with the two HDMI ports connected.

Also make the two HDMI ports use different i.MX6 IPUs.

Signed-off-by: Fabio Estevam <festevam@gmail.com>
---
 arch/arm/boot/dts/imx6qdl-ccimx6sbc.dtsi     | 7 ++++++-
 drivers/gpu/drm/bridge/adv7511/adv7511_drv.c | 2 ++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6qdl-ccimx6sbc.dtsi b/arch/arm/boot/dts/imx6qdl-ccimx6sbc.dtsi
index 2f4ec2545df0..029810a23927 100644
--- a/arch/arm/boot/dts/imx6qdl-ccimx6sbc.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-ccimx6sbc.dtsi
@@ -54,8 +54,13 @@
 	status = "okay";
 };
 
+
+/delete-node/&ipu1_di0_hdmi;
+/delete-node/&hdmi_mux_0;
+/delete-node/&ipu1_di1_hdmi;
+/delete-node/&hdmi_mux_1;
+
 &hdmi {
-	ddc-i2c-bus = <&i2c3>;
 	status = "okay";
 };
 
diff --git a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
index 9e13e466e72c..ccfcf5d3ebca 100644
--- a/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
+++ b/drivers/gpu/drm/bridge/adv7511/adv7511_drv.c
@@ -595,6 +595,8 @@ static int adv7511_get_modes(struct adv7511 *adv7511,
 	struct edid *edid;
 	unsigned int count;
 
+	return 0;
+
 	/* Reading the EDID only works if the device is powered */
 	if (!adv7511->powered) {
 		unsigned int edid_i2c_addr =
-- 
2.17.1

